pipenv/utils/resolver.py

```diff
@@ -630,6 +660,37 @@ def parsed_constraints(self):
             )
         return self._parsed_constraints
 
+    @property
+    def parsed_default_constraints(self):
+        pip_options = self.pip_options
+        pip_options.extra_index_urls = []
+        if self._parsed_default_constraints is None:
+            self._parsed_default_constraints = parse_requirements(
+                self.default_constraint_file,
+                constraint=True,
+                finder=self.finder,
+                session=self.session,
+                options=pip_options,
+            )
+        return self._parsed_default_constraints
+
+    @property
+    def default_constraints(self):
+        from pipenv.patched.pip._internal.req.constructors import (
```

Comment:
Would be better if these imports were at the top of the files -- shouldn't be a circular import in this case.

Reply:
I added that at the top of the file.